Source/ACE/Entity/Landblock.cs

```diff
@@ -467,6 +472,90 @@ private void HandleGameAction(QueuedGameAction action, Player player)
                         HandleSoundEvent(obj, soundEffect);
                         break;
                     }
+                case GameActionType.DropItem:                  
+
+                    {                        
+                        var g = new ObjectGuid(action.ObjectId);                        
+                        // ReSharper disable once InconsistentlySynchronizedField
+                        if (worldObjects.ContainsKey(g))
+                        {                         
+                            var playerId = new ObjectGuid(action.ObjectId);
+                            var inventoryId = new ObjectGuid(action.SecondaryObjectId);
+                            if (playerId.IsPlayer())
+                            {
+                                Player aPlayer = null;
+                                WorldObject inventoryItem = null;
+                                lock (objectCacheLocker)

```

Comment:
You know, I don't think you actually need to lock up the landblock's mutex here.  You're not modifying the object collection in this code block.